File: readability.c
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void) {
    char words[1000];
    printf("Text: ");
    fgets(words, sizeof(words), stdin);
    int num_letters = 0, num_words = 0, num_sentences = 0;
    for (int i = 0; i < strlen(words); i++) {
        if (words[i] == ' ') {
            num_words++;
        } else if (isalpha(words[i])) {
            num_letters++;
        } else if (words[i] == '.' || words[i] == '?' || words[i] == '!') {
            num_sentences++;
        }
    }
    num_words++;
    float L = (float)num_letters / num_words * 100;
    float S = (float)num_sentences / num_words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1) {
        printf("Before Grade 1\n");
    } else if (index > 16) {
        printf("Grade 16+\n");
    } else {
        printf("Grade %d\n", index);
    }
}

File: readability.py
import re

def get_string():
    while (True):
        try:
            text = input("Text: ")
            return text
        except:
            continue

def main():
    text = get_string()
    # Split on end marks; discard final match (after last end mark).
    sentences = re.split(r'[.!?]+', text)[:-1]
    words = [word for s in sentences for word in s.split()]
    letters = [c for word in words for c in word if c.isalpha()]
    L = len(letters) / len(words) * 100
    S = len(sentences) / len(words) * 100
    level = round(0.0588 * L - 0.296 * S - 15.8)
    if level < 1:
        print("Before Grade 1")
    elif level > 16:
        print("Grade 16+")
    else:
        print(f"Grade {level}")

main()
